packages/jest-runner/src/run_test.js

```diff
@@ -23,29 +23,32 @@ import {
   setGlobal,
 } from 'jest-util';
 import jasmine2 from 'jest-jasmine2';
+import LeakDetector from 'jest-leak-detector';
 import {getTestEnvironment} from 'jest-config';
 import * as docblock from 'jest-docblock';
 
 // The default jest-runner is required because it is the default test runner
 // and required implicitly through the `testRunner` ProjectConfig option.
 jasmine2;
 
-export default (async function runTest(
+// Keeping the core of "runTest" as a separate function (as "runTestHelper") is
+// key to be able to detect memory leaks. Since all variables are local to the
+// function, when "runTestHelper" finishes its execution, they can all be freed,
+// UNLESS someone else is leaking them (and that's why we can detect the leak!).
+//
+// If we had all the code in a single function, we should manually nullify all
+// references to verify if there is a leak, which is not maintainable and error
+// prone. That's why "runTestHelper" CANNOT be inlined inside "runTest".
+async function runTestHelper(
   path: Path,
   globalConfig: GlobalConfig,
   config: ProjectConfig,
   resolver: Resolver,
-) {
-  let testSource;
-
-  try {
-    testSource = fs.readFileSync(path, 'utf8');
-  } catch (e) {
-    return Promise.reject(e);
```

Comment:
Such clowny code.

Reply:
The function wasn't async at the point it was written 🙂